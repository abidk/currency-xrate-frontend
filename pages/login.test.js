import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Login from "./login.js";

describe(__filename, () => {
  it("rendered page should match snapshot", () => {
    const component = renderer.create(<Login />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('login submit should call the login function"', () => {
    const submitHandleCalled = jest.fn();

    const app = shallow(<Login />);
    app.find("form").simulate("submit", { preventDefault: submitHandleCalled });

    expect(submitHandleCalled).toHaveBeenCalled();
  });
});
