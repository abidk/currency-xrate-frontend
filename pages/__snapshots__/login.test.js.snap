// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/abid/Documents/currency-xrate/currency-xrate-frontend/pages/login.test.js rendered page should match snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="main"
        >
          <div
            className="login"
          >
            <div
              className="row justify-content-center"
            >
              <div
                className="col-xs-3"
              >
                <div
                  className="card"
                >
                  <div
                    className="card-body"
                  >
                    <form
                      className="form-login"
                      onSubmit={[Function]}
                    >
                      <div
                        className="form-label-group mb-2"
                      >
                        <input
                          autoFocus={true}
                          className="form-control"
                          onChange={[Function]}
                          placeholder="Email address"
                          type="email"
                          value=""
                        />
                      </div>
                      <div
                        className="form-label-group mb-2"
                      >
                        <input
                          className="form-control"
                          onChange={[Function]}
                          placeholder="Password"
                          type="password"
                          value=""
                        />
                      </div>
                      <button
                        className="btn btn-lg btn-primary btn-block text-uppercase"
                        type="submit"
                      >
                        Sign in
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
